atlp-rwanda/phatom-be-techode
# #TP-22 ft:(resetPasword) forgot password and reset password

### What does this PR do?
Allowing a user to reset password

### Description of Task to be completed

- [x] The driver or operator should be able to receive a reset password link in their email.
- [x] The driver or operator should be able to reset the password( i.e. provide a new password).
- [x] The driver or operator should be able to login with the new password.

### How should this be manually tested?

```
git clone https://github.com/atlp-rwanda/phatom-be-techode
cd phantom-be-techode
git checkout ft-create-reset-password-tp-22
npm install
npm run migrate
npm run dev

//after npm run dev check database connection, if it is successful go ahead and test in postman otherwise start up database

```


### Any background context you want to provide?
When sending email I have used a package called **Nodemailer**
N:B for this to run smoothly on your computer, you need to have:
```
EMAIL_USER=<sender's email>
EMAIL_PASS=<password of sender's email>
SERVICE=<service being used>
``` 
 
### What are the relevant Trello stories?
[TP-22](https://trello.com/c/4mPP2nHp/22-reset-password)

[Finishes #TP-22]
